### PR TITLE
Saga

### PR DESCRIPTION
**Pull request by amerkul**
Account service creates a bid and Kafka producer sends an event with the PENDING status to the Auction application. Auction application returns an UpdateAccountCommand with the ACCEPTED or REJECTED status. The bidder account is updated and a new one event is created. The aggregate and event are stored in the database.

### CODE REVIEW

File: src/main/java/com/example/account/kafka/config/ReactiveKafkaConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Bean
+    public ReactiveKafkaProducerTemplate<String, AccountCreatedEvent> reactiveKafkaProducerTemplate(
+            KafkaProperties properties) {
+        Map<String, Object> props = properties.buildProducerProperties();
+        return new ReactiveKafkaProducerTemplate<>(SenderOptions.create(props));
+    }
+
+    @Bean
+    public ReceiverOptions<String, UpdateAccountCommand> kafkaReceiverOptions(@Value("${spring.kafka.topics.answer}") String topic, KafkaProperties kafkaProperties) {
```
Comment (h1alexbel) [inline src/main/java/com/example/account/kafka/config/ReactiveKafkaConfiguration.java]: line is too long

File: src/main/java/com/example/account/kafka/consumer/ReactiveKafkaConsumer.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.example.account.command.UpdateAccountCommand;
+import com.example.account.service.command.CommandService;
+import jakarta.annotation.PostConstruct;
+import lombok.AllArgsConstructor;
+import org.springframework.kafka.core.reactive.ReactiveKafkaConsumerTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class ReactiveKafkaConsumer {
```
Comment (h1alexbel) [inline src/main/java/com/example/account/kafka/consumer/ReactiveKafkaConsumer.java]: Naming between `Consumer` and `Producer` is not unified

File: src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java [hunk 3]
```diff
@@ -34,4 +36,21 @@ public void createAccount(CreateAccountCommand command) {
...
 
+    @Override
+    public void update(UpdateAccountCommand command) {
+        AccountAggregate accountAggregate = AccountAggregate.builder()
+                                                            .status(command.getStatus())
+                                                            .name(command.getName())
+                                                            .balance(command.getBalance())
+                                                            .reserved(command.getReserved())
+                                                            .auctionId(command.getAuctionId())
+                                                            .build();
```
Comment (h1alexbel) [inline src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: code should be formatted

File: src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java [hunk 4]
```diff
@@ -34,4 +36,21 @@ public void createAccount(CreateAccountCommand command) {
...
+        accountRepository.save(accountAggregate)
+                         .doOnNext(accountUpdated -> {
+                             AccountUpdatedEvent event = new AccountUpdatedEvent();
+                             event.apply(accountAggregate);
+                             eventHandler.handle(event);
+                         }).subscribe();
```
Comment (h1alexbel) [inline src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: don't break the reactive pipeline, don't use `subscribe` in reactive context

Comment (h1alexbel) [reply src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: @amerkul `subscribeOn` can help

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java (hunk 4 above)
Comment (amerkul) [reply src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: Or just return a Mono object via controller.

File: src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java (hunk 3 above)
Comment (amerkul) [reply src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: What do you mean?

Comment (h1alexbel) [reply src/main/java/com/example/account/service/command/impl/CommandServiceImpl.java]: format it up to the left grid

**Review by h1alexbel — APPROVED** [review]

**Pull request merged**